.github/workflows/preview-theme.yml

```
@@ -0,0 +1,73 @@
+name: Preview Theme Changes
+
+on:
+  pull_request:
+
+jobs:
+  check-for-changes-to-themes:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Retrieved Theme Changes
+        id: check-for-changes
+        run: |
+          # Retrieve list of all changed files
+          git fetch origin trunk:trunk
+          changed_files=$(git diff --name-only HEAD origin/trunk)
+          
+          # Loop through changed files and identify parent directories
+          declare -A unique_dirs
+          for file in $changed_files; do
+            dir_name=$(dirname "$file")
+            while [[ "$dir_name" != "." ]]; do
+              if [[ -f "$dir_name/style.css" ]]; then  # Check if the parent directory contains a theme
+                # Save only the basename
+                unique_dirs[$dir_name]=$(basename $dir_name)
+                break
+              fi
+              dir_name=$(dirname "$dir_name")
+            done
+          done
+          # Check if themes have changed
+          if [[ ${#unique_dirs[@]} -eq 0 ]]; then
+              echo "No theme changes detected"
+              echo "HAS_THEME_CHANGES=false" >> $GITHUB_OUTPUT
+              exit 78 # Use neutral exit code
+          fi
+          # Output list of theme slugs with changes
+          echo "HAS_THEME_CHANGES=true" >> $GITHUB_OUTPUT
+          echo "CHANGED_THEMES=$(echo ${unique_dirs[@]})" >> $GITHUB_ENV
+          echo "Theme directories with changes: $CHANGED_THEME_DIRS"
```

```suggestion
          echo "Theme directories with changes: $CHANGED_THEMES"
```

There's a mismatch between the variable names here.
I added that line as debug, but I kept it in because it's nice to have that kind of info in the CI; it's not functionally necessary though, in case you'd like to remove it.

---

good catch! and thank you for the script :D